packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-node.vue

```diff
@@ -106,14 +106,15 @@ const startForecast = async (simulationIntervetions) => {
 		},
 		engine: 'ciemss'
 	};
-	// Explicitly add interventions provided. Interventions within the model config will still be utilized either way
-	// TODO: https://github.com/DARPA-ASKEM/terarium/issues/4025
-	console.log(
-		`We now need to concat this with the policy intervention provided and make an object in TDS ${simulationIntervetions}`
-	);
-	// if (simulationIntervetions) {
-	// 	simulationPayload.interventions = simulationIntervetions;
-	// }
+
+	if (optimizedInterventions) {
+		// Use the intervention policy ID provided.
+		simulationPayload.policyInterventionId = optimizedInterventions.id;
+	} else {
+		// Use the input interventions provided
+		const inputIntervention = props.node.inputs?.[2]?.value?.[0];
```

Comment:
I don't think `props.nodes.inputs` can be null/undefined here according to the definition, do you mind doing a quick look around and clean up?

Reply:
Thank you